### DISCUSSION TITLE
[FR] Add data as array

### DISCUSSION THREAD

**Discussion by yurijserrano** [mdelobelle/metadatamenu#533] [Tips and Tricks]
Good Afternoon @mdelobelle 

I think it would be interesting to add an option to add the data as an array; for example, when inserting this field, we would manually put the values we want to be displayed as an array.

I know there are some options to return the values as an array through DataviewJS, but there is no option to include them manually, and this would be fundamental to cover more use cases.

I wish you a wonderful day.

**Comment by mdelobelle** [comment]
Is the "input" type not suiting your need? You can just input your values separated by a comma for an inline field or into brackets for a frontmatter field, and it will be interpreted by dataview and obsidian as arrays.

**Comment by jpfieber** [comment]
I'm trying this but without success.  I have a frontmatter property called "squats" with a value of "[100,8,8,10]" to represent a workout where I squatted 100 pounds for 3 sets: 8 reps, 8 reps, 10 reps.  I want to have another frontmatter called "squatweight" that adds all the reps, then multiplies by the weight to show the total lifted weight of 2600.  I have a preset field called "squatweight" with the formula "current.squats[1] + current.squats[2] + current.squats[3]", but it returns "100", which is characters 1, 2 and 3, not values 1, 2 and 3.  What am I doing wrong?  Once I sum the reps, how do I multiply by the weight (current.squats[0])?

UPDATE:  I put "squats:: 100,8,8,10" in the content, and then used the formula "(current.squats[1] + current.squats[2] + current.squats[3]) * current.squats[0]" and got the correct result of 2600!  How do I move the squats field into the frontmatter?

**Comment by mdelobelle** [comment]
You could try with 'squats' as an ObjectList field
then 'reps' as a Number field with 'squats' as parent
you could also have 'weight' as a Number field with 'squats' as parent
and maybe 'time' as a Time field with 'squats' as parent

Your frontmatter would look like this

```yaml
---
squats:
  - time: 06:30
    reps: 8
    weight: 100
  - time: 08:45
    reps: 8
    weight: 120
  - time: 09:10
    weight: 100
    reps: 8
---
```

then the formula : `current.squats.reduce((p,c) => p+c.weight*c.reps, 0)`

not tested but hopefully you'll get the idea


EDIT: tested, this works as expected

**Reply by jpfieber** [reply]
Thanks, I'll give that a try
